homeassistant/components/airthings_ble/sensor.py

```diff
@@ -0,0 +1,185 @@
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONCENTRATION_PARTS_PER_BILLION,
+    CONCENTRATION_PARTS_PER_MILLION,
+    LIGHT_LUX,
+    PERCENTAGE,
+    PRESSURE_MBAR,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_BLUETOOTH
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN, VOLUME_BECQUEREL, VOLUME_PICOCURIE
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS_MAPPING_TEMPLATE: dict[str, SensorEntityDescription] = {
+    "radon_1day_avg": SensorEntityDescription(
+        key="radon_1day_avg",
+        native_unit_of_measurement=VOLUME_BECQUEREL,
+        name="Radon 1-day average",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radioactive",
+    ),
+    "radon_longterm_avg": SensorEntityDescription(
+        key="radon_longterm_avg",
+        native_unit_of_measurement=VOLUME_BECQUEREL,
+        name="Radon longterm average",
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radioactive",
+    ),
+    "radon_1day_level": SensorEntityDescription(
+        key="radon_1day_level",
+        name="Radon 1-day level",
+        icon="mdi:radioactive",
+    ),
+    "radon_longterm_level": SensorEntityDescription(
+        key="radon_longterm_level",
+        name="Radon longterm level",
+        icon="mdi:radioactive",
+    ),
+    "temperature": SensorEntityDescription(
+        key="temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        name="Temperature",
+    ),
+    "humidity": SensorEntityDescription(
+        key="humidity",
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        name="Humidity",
+    ),
+    "pressure": SensorEntityDescription(
+        key="pressure",
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=PRESSURE_MBAR,
+        name="Pressure",
+    ),
+    "battery": SensorEntityDescription(
+        key="battery",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        name="Battery",
+    ),
+    "co2": SensorEntityDescription(
+        key="co2",
+        device_class=SensorDeviceClass.CO2,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_MILLION,
+        name="co2",
+    ),
+    "voc": SensorEntityDescription(
+        key="voc",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_PARTS_PER_BILLION,
```

Comment:
The volatile organic compounds device class should have ug/m3 as unit.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes